libsolidity/analysis/ControlFlowGraph.h

```diff
@@ -57,8 +57,8 @@ class VariableOccurrence
 	{
 		if (m_occurrence && _rhs.m_occurrence)
 		{
-			if (m_occurrence->id() < _rhs.m_occurrence->id()) return true;
```

Comment:
Does it make sense to add an assertion that they do not contain each other or that either `<` is true or `>`?

Reply:
Why? I think it's better to allow this to compare as equal - it just defines a determinisitic order of occurrences, not sure why it would impose any restrictions on them.

Reply:
Ah, the `SourceLocation::operator<` actually compares the `(start, end)` tuple as is and does not consider inclusion, so it's fine.